lms/djangoapps/courseware/tests/test_tabs.py

```diff
@@ -1,31 +1,216 @@
 """
 Test cases for tabs.
-Note: Tests covering workflows in the actual tabs.py file begin after line 100
 """
+
 from django.conf import settings
 from django.core.urlresolvers import reverse
 from django.http import Http404
 from mock import MagicMock, Mock, patch
-from nose.plugins.attrib import attr
 from opaque_keys.edx.locations import SlashSeparatedCourseKey
 
 from courseware.courses import get_course_by_id
+from courseware.tabs import (
+    get_course_tab_list, CoursewareViewType, CourseInfoViewType, ProgressCourseViewType,
+    StaticCourseViewType, ExternalDiscussionCourseViewType, ExternalLinkCourseViewType
+)
 from courseware.tests.helpers import get_request_for_user, LoginEnrollmentTestCase
 from courseware.tests.factories import InstructorFactory, StaffFactory
-from xmodule import tabs
-from xmodule.modulestore.tests.django_utils import (
-    TEST_DATA_MIXED_TOY_MODULESTORE, TEST_DATA_MIXED_CLOSED_MODULESTORE
-)
-
-from courseware.tabs import get_course_tab_list
 from courseware.views import get_static_tab_contents, static_tab
+from student.models import CourseEnrollment
 from student.tests.factories import UserFactory
 from util import milestones_helpers
+from xmodule import tabs as xmodule_tabs
+from xmodule.modulestore.tests.django_utils import (
+    TEST_DATA_MIXED_TOY_MODULESTORE, TEST_DATA_MIXED_CLOSED_MODULESTORE
+)
 from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
 from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
 
 
-@attr('shard_1')
```

Comment:
Good catch. I did that because I was having odd test failures and I thought we'd moved these tests in from common where the shards are different. I'll add them back.